### ISSUE TITLE
Problem with certain languages and character encoding.

### ISSUE THREAD

**Issue by adityashedge** [wkhtmltopdf/wkhtmltopdf#3170]
I am generating images from HTML using `wkhtmltoimage` and my HTML contains UTF-8 characters in Indian local languages like Hindi and Marathi.
`wkhtmltoimage` modifies the UTF-8 text and the resultant text is incorrect so `दिन` becomes `दनि`.
`दिन` is composed of characters `द`, `ि` and `न` and as per the rules the second half character(`ि`) is appended to the first character but `wkhtmltoimage` appends it to the second character or just appends all the characters to form a string without applying any rules.

If you want to test this the run `wkhtmltoimage http://www.bbc.com/hindi test.jpg`
Thank you in advance for helping me out.

**Comment by PhilterPaper** [comment]
I think the first task would be to find out if it's WebKit that is failing to properly render Hindi text, or if the problem is elsewhere. Does it work correctly in current WebKit engine browsers? If so, is wkHTMLto* using a back-level version of WebKit? If not, does the Chromium engine work better? There is a discussion (#3163) about switching engines that might be of interest.

**Comment by adityashedge** [comment]
@PhilterPaper WebKit is rendering the Hindi text correctly. I was having this issue on my Mac OSX but I noticed that this issue did not appear on my colleague's Ubuntu. Turns out he had additional fonts installed for Devanagiri(Hindi, Marathi-like languages). I installed the `ttf-devanagari-fonts` on AWS Ubuntu Trusty(14.04) instance and the images were generated correctly.

TL;DR I might be a system font issue but then the Safari browser renders it correctly. It is not working correctly on OSX but it is working correctly on Ubuntu Trusty(14.04). Can someone confirm so we can close this issue.
